navigation/src/androidMain/kotlin/com/freeletics/khonshu/navigation/NavHost.kt

```diff
@@ -134,12 +134,13 @@ private fun SystemBackHandling(snapshot: StackSnapshot, executor: MultiStackNavi
 @Composable
 private fun DestinationChangedCallback(
     snapshot: StackSnapshot,
-    destinationChangedCallback: ((BaseRoute) -> Unit)?,
+    destinationChangedCallback: ((NavRoot, BaseRoute) -> Unit)?,
 ) {
     if (destinationChangedCallback != null) {
+        val root = snapshot.root
         val current = snapshot.current
-        DisposableEffect(current) {
-            destinationChangedCallback(current.route)
+        DisposableEffect(root, current) {
```

Comment:
I think `destinationChangedCallback` should be added to the key list, or we use `rememberUpdatedState` instead 🙏

Reply:
Thanks for catching this [e8d2abc](https://github.com/freeletics/khonshu/pull/769/commits/e8d2abc3e149a083704d0733296c2bbacb295b23)